File: Backend/backend/src/main/java/com/example/crediSense/Service/DossierService.java
package com.example.crediSense.Service;

import java.util.List;
import java.util.UUID;

import com.example.crediSense.dto.request.DossierRequest;
import com.example.crediSense.dto.response.DossierResponse;

public interface DossierService {
     DossierResponse create(DossierRequest request);
    DossierResponse getById(UUID id);
    List<DossierResponse> getAll();
    DossierResponse update(UUID id, DossierRequest request);
    void delete(UUID id);
     List<DossierResponse> getByClientId(UUID clientId);
    
}

File: frontend/src/app/services/dossier.service.ts


import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Dossier, DossierRequest } from '../models/dossier.model';

@Injectable({ providedIn: 'root' })
export class DossierService {

  private readonly apiUrl = `${environment.apiUrl}/api/dossiers`;

  constructor(private http: HttpClient) {}

  getByClientId(clientId: string): Observable<Dossier[]> {
    return this.http.get<Dossier[]>(`${this.apiUrl}/client/${clientId}`);
  }

  getById(id: string): Observable<Dossier> {
    return this.http.get<Dossier>(`${this.apiUrl}/${id}`);
  }

  create(request: DossierRequest): Observable<Dossier> {
    return this.http.post<Dossier>(this.apiUrl, request);
  }

  delete(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
